services/wfm/src/main/java/org/openkilda/wfm/share/bolt/AbstractTick.java

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.share.bolt;
+
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.CommandContext;
+
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.apache.storm.utils.TupleUtils;
+
+import java.util.Map;
+
+public abstract class AbstractTick extends AbstractBolt {
```

Comment:
do we need that?

Reply:
Yes. It emit ticks as "normal" tuples not as call to some custom method. It allow to not create "separate" handling of time events. Plus it allow to use synchronized time across whole topology.

Derived classes can define custom "tick" divider without need to reimplement transport logic..